baseplate/lib/config.py

```diff
@@ -390,23 +390,40 @@ def tuple_of(text: str) -> Sequence[T]:
 
 
 def DefaultFromEnv(
-    item_parser: Callable[[str], T], default_src: str, fallback: OptionalType[T] = None
+    item_parser: Callable[[str], T], default_src: str, fallback: OptionalType[T] = None,
 ) -> Callable[[str], OptionalType[T]]:  # noqa: D401
     """An option of type T or a default.
 
     The default is sourced from an environment variable with the name specified in ``default_src``.
     If the environment variable is not set, then the fallback will be used.
     One of the following values must be provided: fallback, default_src, or the provided configuration
     """
-    env = os.getenv(default_src) or ""
-    default = Optional(item_parser, fallback)(env)
+    parser = OptionalDefaultFromEnv(item_parser, default_src, fallback)
 
     def default_from_env(text: str) -> OptionalType[T]:
-        val = Optional(item_parser, default)(text)
+        val = parser(text)
         if val:
             return val
 
-        raise ValueError("no value provided")
+        raise ValueError("Value is required.")
```

Comment:
can this stay how it was before please? just for consistency with other error messaging in this file. (it's not a whole sentence, it gets concatenated into one)